src/Appwrite/Auth/Validator/PasswordNew.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace Appwrite\Auth\Validator;
+
+use Appwrite\Auth\Auth;
+use Utopia\Validator;
+use Utopia\Database\Document;
+
+class PasswordNew extends Validator
+{
+    protected ?Document $project;
+    
+    protected ?Document $user;
+
+    protected array $dictionary;
+
+    protected bool $strict;
+
+    protected string $description = 'Password must be at least 8 characters';
+
+    public function __construct(
+        Document $project = null, 
+        Document $user = null, 
+        array $dictionary = [], 
+        bool $strict = false
+    ) {
+        $this->project = $project;
+        $this->user = $user;
+        $this->dictionary = $dictionary;
+        $this->strict = $strict;
+    }
+
+    public function getDescription(): string
+    {
+        return $this->description;
+    }
+
+    public function isValid($value): bool
+    {
+        /* Base password rules */
+        if (!\is_string($value) || \strlen($value) < 8) {
+            $this->description = 'Password must be at least 8 characters';
+            return false;
+        }
+
+        /* Dictionary check */
+        if ($this->project && !$this->project->isEmpty()) {
+            if (!$this->isValidDictionaryCheck($this->project, $value)) {
+                $this->description = 'Password should not be one of the commonly used passwords';
+                return false;
+            }
+        }
+
+        if ($this->project && !$this->project->empty() && $this->user && !$this->user->empty()) {
+            /* History check */
+            if (!$this->isValidHistoryCheck($this->project, $this->user, $value)) {
+                $this->description = 'Password shouldn\'t be in the history.';
```

Comment:
Same, what history? Maybe something like previously used passwords are not allowed?

Reply:
Understood 👍  I will update accordingly. For now I just copied what we were already using